### ISSUE TITLE
Add support for custom plugins

### ISSUE THREAD

**Issue by berestovskyy** [WMEValidator/validator#58]
The unique feature of Validator is its report system. The idea is to add an open interface for other scripts, similar to what we have for localization packs. Those scripts could be tiny, without any user interface (just like localization packs are). Validator will trigger those scripts for each segment/place, scripts will check the segments/places and report back issues.

The example user workflow:

1. Install Validator.
2. Install few custom plugins.
3. Use Validator as usual.
4. Validator will show all plugin's reports and report them to user.

Long term benefits:

1. This will make Validator much smaller by extracting all the current checks as plugins.
2. This will make a bridge between Validator and other (existing) scripts, so we don't need to reimplement existing checks.
3. Hopefully, other script writers will be more interested in contributing to Validator or its plugins, because at the moment other script writers might treat Validator as a competitor.

**Comment by berestovskyy** [comment]
Historically Validator always tried its best to cooperate with other popular scripts. If Toolbox of Color Highlight were installed alongside Validator, the report generated by Validator included issues reported by those scripts.

The plugin system is a step further with the integration. It is a bit of work, though...

**Comment by davidakachaos** [comment]
I agree that a plugin system would be awesome to work with. Much better that adding for example places checks like I suggested in #57 to the core script.
I'll have a look around at other scripts that implement a sort of plugin system (Wide Angle Lens comes to mind) Maybe we could use ideas in those scripts for adding a plugin system...
